### PR TITLE
DEV - Set traitlets application version

### PR DESCRIPTION
**Pull request by soapy1**
Fixes https://github.com/conda-incubator/conda-store/issues/977

## Description

This pr set's up the traitlets application to know the current version of conda-store. Setting the version property of a traitlets Application will allow the built in trailtlets '--version' flag to work properly.

For example, running `conda-store-server --version` now works properly
```
 $ conda-store-server --version 
2024.11.1
 $ conda-store-worker --version
2024.11.1
```

ref:
  https://github.com/ipython/traitlets/blob/main/traitlets/config/application.py#L872
  https://github.com/ipython/traitlets/blob/main/traitlets/config/application.py#L200

## Pull request checklist

- [x] Did you test this change locally?
- [x] Did you update the documentation (if required)?
- [x] Did you add/update relevant tests for this change (if required)?

### CODE REVIEW

**Comment by soapy1** [comment]
The `Build and maybe upload PyPI package /...` is currently a little broken, but fixed in [this pr ](https://github.com/conda-incubator/conda-store/pull/980).  Will rebase this pr to pick up the fix once #980 is merged

**Review by trallard — APPROVED** [review]
Thank you, this LGTM 🚀 will approve and wait for the CI to complete.

**Pull request merged**
